### DISCUSSION TITLE
R - Nilai pada row tidak tampil (bukan NA), namun memiliki nilai ketika di subset

### DISCUSSION THREAD

**Discussion by Talioll** [teamalgoritma/community#409] [Q&A] [answered]
CASE :

1.  Kolom merupakan data bertipe faktor dengan unique value : "village", "city", "town", dan ""
2.  Ketika menampilkan dataframe, data dengan row "" tidak muncul (krn memang [" "] diartikan kosong/namun bukan NA)
3.  Ketika di subset, row dengan value [" "] ternyata memiliki nilai (nilai dapat berupa village"/"city"/"town")
4.  Nilai [" "] yang muncul tidak memiliki pola yang pasti (saya pernah mendapat kasus seperti ini juga, namun pada kasus yg pernah sy tangani nilai yang tidak tampil tersebut memiliki pola yang sama yaitu mengikuti nilai row diatasnya. Namun pada kasus ini tidak seperti itu, sy sudah melakukan cek nilai row atas dan bawah dari row [" "], namun hasilnya bisa sama dan bisa berbeda dengan row di atas maupun di bawah sehingga kesimpulannya nilai tersebut tidak bergantung dengan row sebelum/setelahnya).
5. Bagaimana penyelesaian yang dapat dilakukan?

Terima kasih :)

![Question](https://user-images.githubusercontent.com/112487401/197784148-b7b90335-cdea-4553-833c-ddd0ab358eab.jpg)

**Comment by onlyphantom** [comment, marked as answer]
Actually it's because you've confused yourself with treating `X` as the row index. So a few examples:

```
# this returns all rows where region is empty ("")
subset(churn, region == "")

# this returns row number 37, and region of that row 37 is not empty
churn[37, "region"]

# this returns row where X is 37, and region of that row is empty
churn[churn[X] == 37, "region"]
```

Secara kesimpulan, sumber confusion nya adalah `X` kolum itu tidak sama dengan row index nya. Mungkin bisa experimentasi juga:
```
churn[37, c("X", "region")]
```
Dan coba merperhatikan bahwa X itu tidak sama dengan row index nya. Jadi kalo subset nya pakai `churn[37, "region"]` itu tidak sama dengan `churn[churn$X == 37, "region"]`.

Semoga membantu. Good luck!

**Reply by Talioll** [reply]
betul setelah sy post ini sy juga sadar adalah salah indexing :(
